src/test/java/kitchenpos/bo/TableGroupBoTest.java

```diff
@@ -0,0 +1,159 @@
+    void create_orderTable_size() {
+        final TableGroup tableGroup = new TableGroup();
+        tableGroup.setId(1l);
+
+        final OrderTable orderTable = new OrderTable();
+        orderTable.setId(1l);
+        orderTable.setEmpty(true);
+
+
+        final OrderTable orderTable1 = new OrderTable();
+        orderTable1.setId(2l);
+        orderTable1.setEmpty(true);
+
+        final List<OrderTable> orderTables = Arrays.asList(orderTable, orderTable1);
+        tableGroup.setOrderTables(orderTables);
+
+        final List<Long> orderTableIds = orderTables.stream().map(OrderTable::getId).collect(Collectors.toList());
+
+        given(orderTableDao.findAllByIdIn(orderTableIds)).willReturn(Arrays.asList(orderTable));
+
+        assertThatExceptionOfType(IllegalArgumentException.class)
+                .isThrownBy(() -> tableGroupBo.create(tableGroup));
+    }
+
+    @Test
+    @DisplayName("테이블 그룹을 삭제 할 수 있다.")
+    void delete() {
+        final TableGroup tableGroup = new TableGroup();
+        tableGroup.setId(1l);
+        tableGroup.setOrderTables(orderTables);
+
+        final List<Long> orderTableIds = orderTables.stream().map(OrderTable::getId).collect(Collectors.toList());
+
+        final List<OrderTable> result = Fixtures.orderTables;
+        result.forEach(res -> res.setTableGroupId(null));
+
+        given(orderTableDao.findAllByTableGroupId(tableGroup.getId())).willReturn(Fixtures.orderTables);
+        given(orderDao.existsByOrderTableIdInAndOrderStatusIn(orderTableIds,
+                Arrays.asList(OrderStatus.COOKING.name(), OrderStatus.MEAL.name()))).willReturn(false);
+        given(orderTableDao.save(Fixtures.orderTables.get(0))).willReturn(result.get(0));
+        given(orderTableDao.save(Fixtures.orderTables.get(1))).willReturn(result.get(1));
+        given(orderTableDao.save(Fixtures.orderTables.get(2))).willReturn(result.get(2));
+
+        tableGroupBo.delete(tableGroup.getId());
+
+        assertThat(result.get(0).getTableGroupId()).isNull();
+        assertThat(result.get(1).getTableGroupId()).isNull();
+        assertThat(result.get(2).getTableGroupId()).isNull();
+    }
+
+    @Test
+    @DisplayName("주문 상태가 요리중이거나 먹고있으면 삭제 할 수 없다.")
+    void delete_OrderStatus_cooking_or_meal() {
+        final TableGroup tableGroup = new TableGroup();
+        tableGroup.setId(1l);
+        tableGroup.setOrderTables(orderTables);
+
+        final List<Long> orderTableIds = orderTables.stream().map(OrderTable::getId).collect(Collectors.toList());
+
+        final List<OrderTable> result = Fixtures.orderTables;
+        result.forEach(res -> res.setTableGroupId(null));
+
+        given(orderTableDao.findAllByTableGroupId(tableGroup.getId())).willReturn(Fixtures.orderTables);
+        given(orderDao.existsByOrderTableIdInAndOrderStatusIn(orderTableIds,
+                Arrays.asList(OrderStatus.COOKING.name(), OrderStatus.MEAL.name()))).willThrow(IllegalArgumentException.class);
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
                Arrays.asList(OrderStatus.COOKING.name(), OrderStatus.MEAL.name()))).willReturn(true);
```

Reply:
given 조건이라서 throw exception 을 날리는게 아니라,  명시적으로 표시했어야 했네요